# Exercise 1 - Count with python

# 0. Pythagorean theorem (*)
a) A right angled triangle has the catheti: a = 3 and b = 4 length units. Compute the hypothenuse of the triangle. (*)

b) A right angled triangle has hypothenuse c = 7.0 and a cathetus a = 5.0 length units. Compute the other cathetus and round to one decimal. (*)


In [3]:
#a roten ur 
import math

a = 3
b = 4 

c = math.sqrt(a**2 + b**2)
print(f"Hypotenusa is {c}")



Hypotenusa is 5.0


In [4]:
#b 
#Hypo
c = 7.0 
#cathetus 
a = 5.0 

b = math.sqrt(c**2 - a**2)

print(f"{b:.1f}")


4.9


# 1. Classification accuracy (*)
A machine learning algorithm has been trained to predict whether or not it would rain the next day. Out of 365 predictions, it got 300 correct, compute the accuracy of this model.

In [11]:
predictions = 365
correct_pre = 300 

accuracy = correct_pre / predictions * 100 

print(f"The accuracy of the calculator is {accuracy:.1f} %")


The accuracy of the calculator is 82.2 %


# 2. Classification accuracy (*)

A machine learning model has been trained to detect fire. Here is the result of its predictions:

Calculated the accuracy using the following formula:


Is this a good model? Why? Why not?

In [16]:
TP = 2 
FP = 2 
FN = 11 
TN = 985

accuracy_2 = (TP + TN) / (TP + TN + FP + FN) * 100 

print(f"{accuracy_2:.1f} %")

98.7 %


# 3. File metadata - format print (*)
There is a file with the following metadata:

name: orders_data.csv
file size: 3.14mb
modified 2024
Create variables for each of them and print out the following to the screen

In [27]:
filename = "orders_data.csv"
file_size = 3.14
modiefied = 2024

print(f"=== File metadata === \n Filename: {filename} \n Size (MB): {file_size} \n Last Modidied Year: {modiefied} ")




=== File metadata === 
 Filename: orders_data.csv 
 Size (MB): 3.14 
 Last Modidied Year: 2024 


# 4. Unit converter - inch to cm (*)
Screen sizes are defined using inches for example my computer has 15'' and it corresponds to the diagonal of the screen. Let the user type in a value in inches and output the result in centimeters.

In [31]:
inchesInput = int(input("Enter the number in inches that you want to convert to cm"))

inchesInCm = inchesInput * 2.54

print(f"{inchesInput} inches is {inchesInCm:.1f} cm")

45 inches is 114.3 cm


# 5. Count characters in a text (*)
Create a program to let a user input a text, count the number of characters and print it out to the screen. Space can also be counted as a character here.

Hint: len() function

In [37]:
userText = input("Put the text in that you want to be counted")

countText = len(userText)

print(countText)


2


# 6. Data types (*)
Create variables of type bool, int, float, str and list. Then use type function and check each variables type.

In [59]:
boolT = False
intN = 3
floatN = 20.201
stringTxt = "Miranda"
fruits = ["apple", "pear", "cherry"]

#Self-documenting expression with = operator in f-string 
print(f"{boolT = }. {boolT} has type {type(boolT)}")
print(f"{intN = }. {floatN} has type {type(floatN)}")
print(f"{stringTxt = }. {stringTxt} has type {type(stringTxt)}")
print(f"{fruits = }. {fruits} has type {type(fruits)}")

boolT = False. False has type <class 'bool'>
intN = 3. 20.201 has type <class 'float'>
stringTxt = 'Miranda'. Miranda has type <class 'str'>
fruits = ['apple', 'pear', 'cherry']. ['apple', 'pear', 'cherry'] has type <class 'list'>


# 7. Theory
a) What is the difference between input() and print() in Python?

Input = function to read the input from a user 
print =  function to print variables etc 

b) What data type does the input() function return by default? How can you use it to get numeric input?

Input returns string automaticlly. 
to return an int you need to convert the string to an int by input(int())

c) Explain how formatted string literals (f-strings) improve readability in output statements. Give an example.

The f-string formatter is very good when you want to display variables or use some matematics or special caracthers 
 
print(f"{10*10}")

name = "Miranda"
ex1 print("hej jag heter Miranda")
ex2 print(f"hej jag heter {name}")


d) Can a variable change its data type during runtime in Python? Show an example.

Yes you can assign a new value to the variable and for example make it an int instead of an float. 

floatToInt = 20.20
floatToInt = 10 


You can also change it by using functions 
floating = 10.4


print(f"{str(floating)}")

e) What does it mean that Python is "dynamically typed"? How is that different from statically typed languages?

It means that the data type is associated with the value and not the variable. therefor you don't have to declaire the data type. It gives better syntax, flexibikity, faster development etc.  
